### PR TITLE
Node js w3 homework

### PR DESCRIPTION
**Pull request by gizemcandemir**

### CODE REVIEW

File: nodejs/week3/warmup/index.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+	} else if (method === "division") {
+		return data.reduce((sum, value) => (sum = sum / value), 1);
+	} else {
+		return `Method is not recognized`;
+	}
+}
+
+// calculator/add?firstParam=1&secondParam=2
+app.get('/calculator/add', (req, res) => {
+	const queryData = [].concat.apply([], Object.values(req.query));
```
Comment (benna100) [inline nodejs/week3/warmup/index.js]: Where do you have this line from? I have seen it in multiple homeworks.

File: nodejs/week3/warmup/index.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+	const data = queryData.map(Number);
+	const result = data.reduce((sum, value) => (sum = sum + value), 0);
```
Comment (benna100) [inline nodejs/week3/warmup/index.js]: Great with the reduce. Not so great with the variable name. What would be a better name?

File: nodejs/week3/warmup/index.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+	console.log(data[0]);
+	const result = data.reduce((sum, value) => (sum = sum - value));
+	res.send(`Query params: ${queryData} and the result is: ${result}`);
+})
+
+// calculator/multiply?firstParam=1&secondParam=2&secondParam=4
+app.get('/calculator/multiply', (req, res) => {
+	const queryData = [].concat.apply([], Object.values(req.query));
+	const data = queryData.map(Number);
+	const result = data.reduce((sum, value) => (sum = sum * value), 1);
```
Comment (benna100) [inline nodejs/week3/warmup/index.js]: `queryData`, `data` and `result` are all examples i use of vague variable names. The reason why you should avoid them is that they are vague and dont add any information about the variable. Try when you give variable names to imagine explaining your mom about the variable. What would you say to her the variable contains? Use that! **Explain what the variable contains, not what it is**

File: nodejs/week3/warmup/index.js (hunk 1 above)
Comment (gizemcandemir) [reply nodejs/week3/warmup/index.js]: Some of us came together for a studygroup at HYF when our class was cancelled before christmas break. We found out that this line works for our homework so I guess we all used it 🙈

**Pull request merged**